sktime/forecasting/statsforecast.py

```diff
@@ -628,14 +644,165 @@ def get_test_params(cls, parameter_set="default"):
+            "season_length": [3, 12],
+            "trend_forecaster": ThetaForecaster(),
+        }
+
+        return params
+
+
+class StatsForecastBackAdapterMSTL(BaseForecaster):
+    """StatsForecast Back Adapter for StatsForecast MSTL.
+
+    StatsForecastBackAdapter is a wrapper for
+
+    Parameters
+    ----------
+    estimator :
+
+    Examples
+    --------
+    >>> from sktime.datasets import load_airline
+    >>> from sktime.forecasting.statsforecast import StatsForecastMSTL
+    >>> from sktime.forecasting.statsforecast import StatsForecastBackAdapterMSTL
+    >>> from sktime.forecasting.theta import ThetaForecaster
+
+    >>> y = load_airline()
+    >>> trend_forecaster = StatsForecastBackAdapterMSTL(ThetaForecaster())
+    >>> model = StatsForecastMSTL(
+            season_length=[3,12],
+            trend_forecaster=trend_forecaster
+        )
+    >>> fitted_model = model.fit(y=y)
+    >>> y_pred = fitted_model.predict(fh=[1,2,3])
+    >>> y_pred
+    1961-01    454.507079
+    1961-02    419.341442
+    1961-03    477.373918
+    Freq: M, Name: Number of airline passengers, dtype: float64
+    """
+    def __init__(self, estimator):
+        self.estimator = estimator
+        self._forecaster = None
+        self._model = None
+        self.in_sample_fh = None
+        self.in_sample_X = None
+
+        super().__init__()
+
+        if hasattr(self.estimator, "seasonal"):
+            if self.estimator.seasonal:
+                raise Exception(
+                    "Trend forecaster should not adjust seasonal models."
+                )
+
+        if hasattr(self.estimator, "sp"):
+            if self.estimator.sp > 1:
+                raise Exception(
+                    "Trend forecaster should not adjust seasonal models. Please pass "
+                    "`sp=1` to your trend forecaster"
+                )
+
+    def __repr__(self):
+        return "StatsForecastBackAdapterMSTL"
+
+    def fit(self, y, X=None):
+        self.estimator.fit(y=y, X=X)
+        self.in_sample_fh = -np.arange(len(y))
+        return self
+
+    def predict(self, h=None, X=None, level=None):
+        mean = self.estimator.predict(fh=range(1, h+1), X=X)[:, 0]
+        if level is None:
+            return {"mean": mean}
+
+        coverage = [round(1-(_l/100), 2) for _l in level]
+
+        pred_int = self.estimator.predict_interval(
+            fh=range(1, h+1),
+            X=X,
+            coverage=coverage
+        )
+
+        return {
+            "mean": mean,
+            **{f"lo-{_l}":
+                pred_int[("Coverage", c, "lower")].values
+                for c, _l in zip(reversed(coverage), level)},
+            **{f"hi-{_l}":
+                pred_int[("Coverage", c, "upper")].values
+                for c, _l in zip(coverage, level)},
+        }
+
+    def predict_in_sample(self, level=None):
+        fitted = self.estimator.predict(
+            fh=self.in_sample_fh,
+            X=self.in_sample_X
+        )[:, 0]
```

Comment:
When testing on `load_airline` dataset, `fitted` is way off (first value is -68, when it should be 112), because `ThetaForecaster` computes a negative drift. Drift should probably positive due to positive trend of airline dataset?